pkg/util/json/patch.go

```diff
@@ -30,6 +30,9 @@ func CreatePatch(before, after interface{}) ([]byte, error) {
 	if err != nil {
 		return nil, err
 	}
+	if len(patch) == 0 {
```

Comment:
ahh, you modified the code here as well to return nil. Given that this is supposed to be a generic function, I am not sure whether it makes more sense to handle the empty JSON diff "[]" higher up.